src/main/java/io/gingersnapproject/rest/EngineResource.java

```diff
@@ -2,23 +2,35 @@
 
 import javax.inject.Inject;
 import javax.ws.rs.GET;
+import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 
-import io.gingersnapproject.cdc.ManagedEngine;
+import io.gingersnapproject.cdc.DynamicRuleManagement;
+import io.gingersnapproject.cdc.RuleLifecycleManagement;
+import io.gingersnapproject.cdc.configuration.DefaultRuleParser;
+
 import org.jboss.resteasy.reactive.RestQuery;
 
-@Path("/v1/engine")
+@Path("/v1/rules")
 public class EngineResource {
 
-   @Inject ManagedEngine managedEngine;
+   @Inject RuleLifecycleManagement engineLifecycleManagement;
+   @Inject DynamicRuleManagement dynamicEngineManagement;
+
+   @POST
+   @Path("{name}")
+   public void addNewEngine(String name, String body) {
+      dynamicEngineManagement.addRule(name, DefaultRuleParser.parse(body));
+   }
 
    @GET
    @Path("{name}")
-   public void enableDisableEngine(String name, @RestQuery("enable") Boolean enable) {
-      if (enable) {
-         managedEngine.start(name);
-      } else {
-         managedEngine.stop(name);
+   public void removeEngine(String name, @RestQuery("action") String action) {
```

Comment:
To delete a rule, I think a `DELETE` `/v1/rules/{name}` operation is more RESTful.